### PR TITLE
updating field to make it nullable

### LINKED ISSUE #5981: Batch2WorkChunkEntity poll_attempts is primitive in a nullable column
Adding a primitive nullable column to an existing table (without migration and downtime) will result in PropertyAccessException's.

This field should be made nullable.

### PR DESCRIPTION
**Pull request by TipzCM**
closes #5981

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/entity/EntityTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+			for (Field field : clazz.getDeclaredFields()) {
+				Column column = field.getAnnotation(Column.class);
+				if (column == null) {
+					continue;
+				}
+
+				if (column.nullable()) {
+					if (field.getType().isPrimitive()) {
+						if (!(CLASS_TO_FIELD_NAME_EXCEPTIONS.containsKey(className)
+							&& CLASS_TO_FIELD_NAME_EXCEPTIONS.get(className).contains(field.getName()))) {
```
Comment (volodymyr-korzh) [inline hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/entity/EntityTest.java:69]: Nitpick: All if condition could be gathered in one if statement, to improve readability:
For example: 
`if (column.nullable() && field.getType().isPrimitive() && isExceptionField(field, className))`

And you can Extract last if statement as a separate method `isExceptionField`

**Review by volodymyr-korzh — APPROVED** [review]

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/entity/EntityTest.java (hunk 1 above)
Comment (TipzCM) [reply hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/entity/EntityTest.java:69]: i'll extract it to a method

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/entity/Batch2WorkChunkEntity.java [hunk 2]
```diff
@@ -352,7 +352,10 @@ public void setNextPollTime(Date theNextPollTime) {
 		myNextPollTime = theNextPollTime;
 	}
 
-	public int getPollAttempts() {
+	public Integer getPollAttempts() {
+		if (myPollAttempts == null) {
+			myPollAttempts = 0;
+		}
 		return myPollAttempts;
```
Comment (tadgh) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/entity/Batch2WorkChunkEntity.java:359]: recommendation: just return 0 if null, i wouldnt' modify it in place. it breaks least astonishment to have a getter modify an entity value.

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/entity/EntityTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+				if (isPrimitiveFieldNullable(className, field, column)) {
+					fail(String.format("Column %s on Entity %s is nullable, but has been defined as primitive.",
+						field.getName(), className)
+						+ " If this is a new field on an existing Table, this can result in deserialization issues when reading old data."
+						+ " Either change to a nullable type (Object) or ensure existing data is migrated and add as an exception.");
+				}
+			}
+		}
+	}
```
Comment (tadgh) [inline hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/entity/EntityTest.java:69]: How long does this test take to run on your machine?

**Review by tadgh — APPROVED** [review]
approved pending questions. Also, why is it nullable? is zero not a sufficient default?

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/entity/Batch2WorkChunkEntity.java (hunk 2 above)
Comment (TipzCM) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/entity/Batch2WorkChunkEntity.java:359]: updated

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/entity/EntityTest.java (hunk 3 above)
Comment (TipzCM) [reply hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/entity/EntityTest.java:69]: Not counting build time?

half a second
image

**Pull request merged**